src/test/unit/model/indexing/util.hpp

```diff
@@ -156,9 +191,11 @@ auto generate_linear_vector(Eigen::Index n, double start = 0) {
  * @param n Number of cells.
  * @param start Where the linear sequence should start from.
  */
-template <typename Vec = Eigen::Matrix<double, -1, 1>>
+template <typename Vec = Eigen::Matrix<double, -1, 1>,
+          typename RhsScalar = stan::math::var>
 auto generate_linear_var_vector(Eigen::Index n, double start = 0) {
-  using ret_t = stan::math::var_value<Vec>;
+  using ret_t = std::conditional_t<is_var<RhsScalar>::value,
+                                   stan::math::var_value<Vec>, Vec>;
```

Comment:
The name of this function is now confusing, as depending in template it does not generate a var vector. Same for the matrix one.

Reply:
I agree, I changed the name to just `conditionally_generate_linear_var_vector()` etc.